docs/build/guides/archival/choosing-the-right-storage.mdx

```diff
@@ -0,0 +1,151 @@
+
+Like temporary storage, persistent storage also has an unlimited amount of storage space. However, it is expensive since the data is stored on the network for a longer time.
+
+Examples of data that may be stored on persistent storage include user balances, token metadata, voting and governance decisions, and all data that need to remain accessible across multiple contract invocations and potentially long durations.
+
+Let's look at a contract for a loyalty points system where users can accumulate points and redeem them for rewards. Each user's point balance will be stored in persistent storage.
+
+```rust
+use soroban_sdk::{contractimpl, contracttype, Address, Env};
+
+#[contracttype]
+pub enum DataKey {
+    Points(Address),
+}
+
+#[contract]
+pub struct LoyaltyPointsContract;
+
+#[contractimpl]
+impl LoyaltyPointsContract {
+    // This function adds points to the user's balance
+    pub fn add_points(env: Env, user: Address, points: u64) {
+        let key = DataKey::Points(user.clone());
+        let current_points: u64 = env.storage().persistent().get(&key).unwrap_or(0);
+        let new_points = current_points + points;
+        env.storage().persistent().set(&key, &new_points);
+    }
+
+    // This function retrieves the user's points balance
+    pub fn get_points(env: Env, user: Address) -> u64 {
+        let key = DataKey::Points(user);
+        env.storage().persistent().get(&key).unwrap_or(0)
+    }
+
+    // This function redeems points according to the user's balance
+    pub fn redeem_points(env: Env, user: Address, points: u64) -> bool {
+        let key = DataKey::Points(user.clone());
+        let current_points: u64 = env.storage().persistent().get(&key).unwrap_or(0);
+
+        if current_points >= points {
+            let new_points = current_points - points;
+            env.storage().persistent().set(&key, &new_points);
+            true
+        } else {
+            false
+        }
+    }
+}
+
+```
+
+### Instance storage
+
+Instance storage is a type of persistent storage that is used for storing data associated with the contract instance itself; data that only needs to be accessed while the contract is active.
+
+Here, the ledger entry shares the same TTL as the contract instance. Therefore once the contract instance is no longer active, the data is archived. However, it can also be restored using the `RestoreFootprintOp` operation.
+
+Like persistent storage, instance storage stores data permanently and for the same fees as persistent storage. The key difference is that instance storage allocates a limited storage space.
+
+Instance storage is best suited for for data that is shared across multiple contract instances but only needs to be accessed during the lifespan of the contract instance. For example, contract admin settings that can be modified by authorized users.
```

Comment:
> shared across multiple contract instances 

Is this true? How can you share across multiple instances but only access from one instance? I don't think storage is ever shared across contracts... :thinking: this might need some clarity